.circleci/config.yml

```diff
@@ -4,59 +4,195 @@ orbs:
+            - yarn-packages-{{ checksum "app/yarn.lock" }}
       - run:
-          name: Install Dependencies
-          command: yarn install --frozen-lockfile
+          name: Install Yarn Dependencies
+          command: |
+            source ~/.bashrc
+            pushd app
+            yarn install --frozen-lockfile
+            popd
       - save_cache:
           name: Save Yarn Package Cache
-          key: yarn-packages-{{ checksum "yarn.lock" }}
+          key: yarn-packages-{{ checksum "app/yarn.lock" }}
           paths:
             - ~/.cache/yarn
       - run:
-          name: Run Tests via Yarn
-          command: yarn test
-      - run:
-          name: Audit Dependencies
-          # https://yarnpkg.com/lang/en/docs/cli/audit/
-          command: yarn audit --level info
-
-  check-code-style:
-    docker:
-      - image: circleci/node:10.14.1
-    working_directory: ~/cas-ciip-portal/app
-    steps:
-      - checkout:
-          path: ~/cas-ciip-portal
+          name: Configure Cpan
+          command: |
+            echo | cpan # accept cpan defaults
       - restore_cache:
-          name: Restore Yarn Package Cache
+          name: Restore Cpan Package Cache
           keys:
-            - yarn-packages-{{ checksum "yarn.lock" }}
+            - cpanm-packages-b-{{ checksum "schema/cpanfile" }}
       - run:
-          name: Install Dependencies
-          command: yarn install --frozen-lockfile
+          name: Install Cpan Dependencies
+          command: |
+            source ~/.bashrc # load perl config (just written by cpan) & asdf
+            cat .tool-versions | xargs -n2 asdf global # make this project's versions global
+            pushd schema
+            make dev_install
+            popd
       - save_cache:
-          name: Save Yarn Package Cache
-          key: yarn-packages-{{ checksum "yarn.lock" }}
+          name: Save Cpan Package Cache
+          key: cpanm-packages-b-{{ checksum "schema/cpanfile" }}
           paths:
-            - ~/.cache/yarn
+            - ~/perl5
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - .asdf
+            - .bashrc
+            - cas-ciip-portal
+            - perl5
+
+  lint:
+    machine:
+      image: ubuntu-1604:201903-01
+    working_directory: ~/cas-ciip-portal/app
+    steps:
+      - attach_workspace:
+          at: ~/
       - run:
           name: Run XO via Yarn
-          command: yarn format || exit 0;
+          command: |
+            source ~/.asdf/asdf.sh
+            yarn format || true # TODO: need to clean up linter errors
       - run:
           name: Check if there is some unformatted code
           command: git diff --exit-code || (echo 'These files are not formatted correctly. Did you forget to \"yarn format\"?' && false)
+      - run:
+          name: Verify style with pre-commit
+          command: |
+            source ~/.asdf/asdf.sh
+            asdf reshim
+            pre-commit run --all-files
```

Comment:
This will fail until we fix all the linter errors. There's only 25 of them, might as well fix them now?

Reply:
Right, of course! Yes, I'll fix...

Reply:
Looks like there are a few that will need a bit of a refactor to fix. Can we do that in a separate PR? Trying to avoid merge conflicts.